airflow/contrib/auth/backends/password_auth.py

```diff
@@ -106,8 +102,8 @@ def load_user(userid, session=None):
     if not userid or userid == 'None':
         return None
 
-    user = session.query(models.User).filter(models.User.id == int(userid)).first()
-    return PasswordUser(user)
+    user = session.query(PasswordUser).filter(PasswordUser.id == int(userid)).first()
```

Comment:
This change looks better, but I need to check it doesn't actually break other things - parts of the code may expect user.user to exist (as bonkers as that sounds)

Reply:
I suspect this will fail tests as https://github.com/apache/incubator-airflow/blob/b7a7fd66d693dbfbc471a6d08bc274441ee4841c/airflow/www/utils.py#L299 won't work anymore.

And this (admittedly silly) API is part of the external API that people have written custom auth backends against so we can't change it. If we were keeping these I'd say it's worth "fixing" this (as you have here) but since we're removing the custom auth backends in Favour of Flask-AppBuilder in 2.0.0 it's probably worth just keeping the sillyness.

Reply:
If it keeps the API from breaking, I can see if I can fix it in [views.py:2070](https://github.com/apache/incubator-airflow/blob/master/airflow/www/views.py#L2070) with something like

```suggestion
    do_filter = FILTER_BY_OWNER and (not current_user.user.is_superuser())
```

Not the nicest way to handle this, I guess, but perhaps worth considering if it enables filtering DAGs by owner for the time being?

Do you have any indication when 2.0.0 is scheduled for release? Depending on the time frame, I may be able to invest a couple of hours to look into this further as this feature would make my life a lot easier :)

Reply:
No timing on 2.0.0 (q1 or q2 next year) but the "RBAC" based UI is available behind a config flag on 1.10.0 so you can start experimenting with it already https://github.com/apache/incubator-airflow/blob/master/UPDATING.md#new-webserver-ui-with-role-based-access-control

Reply:
My use case unfortunately requires DAG level access control. I have tried running master, but the DAG level roles available there do not seem to work on my machine.

I have updated my code to leave the API alone, please let me know if there is anything else I can do to improve this PR.